pkgs/tools/graphics/argyllcms/default.nix

```diff
@@ -25,109 +101,61 @@ stdenv.mkDerivation rec {
     }
   );
 
-  postPatch = lib.optionalString (stdenv.hostPlatform != stdenv.buildPlatform) ''
-    substituteInPlace Jambase \
-      --replace "-m64" ""
-  '';
-
-  preConfigure = let
-    # The contents of this file comes from the Jamtop file from the
-    # root of the ArgyllCMS distribution, rewritten to pick up Nixpkgs
-    # library paths. When ArgyllCMS is updated, make sure that changes
-    # in that file is reflected here.
-    jamTop = writeText "argyllcms_jamtop" ''
-      DESTDIR = "/" ;
-      REFSUBDIR = "share/argyllcms" ;
-
-      # Keep this DESTDIR anchored to Jamtop. PREFIX is used literally
-      ANCHORED_PATH_VARS = DESTDIR ;
-
-      # Tell standalone libraries that they are part of Argyll:
-      DEFINES += ARGYLLCMS ;
-
-      # enable serial instruments & support
-      USE_SERIAL = true ;
-
-      # enable fast serial instruments & support
-      USE_FAST_SERIAL = true ;                # (Implicit in USE_SERIAL too)
-
-      # enable USB instruments & support
-      USE_USB = true ;
-
-      # enable dummy Demo Instrument (only if code is available)
-      USE_DEMOINST = true ;
+  postPatch = ''
+      substituteInPlace Makefile --replace-fail "-j 3" "-j $NIX_BUILD_CORES"
+      substituteInPlace Jamtop --replace-fail "EXIT Unable" "ECHO Unable"
 
-      # enable Video Test Patch Generator and 3DLUT device support
-      # (V2.0.0 and above)
-      USE_VTPGLUT = false ;
+      # Keep udev rules
+      mkdir udev-rules
+      mv usb/*.rules udev-rules
 
-      # enable Printer device support
-      USE_PRINTER = false ;
+      # Remove bundled libraries, making the build fail if not provided externally
+      rm -rf tiff jpeg png usb zlib
 
-      # enable CMF Measurement device and accessory support (if present)
-      USE_CMFM = false ;
+      export AR="$AR rusc"
 
-      # Use ArgyllCMS version of libusb (deprecated - don't use)
-      USE_LIBUSB = false ;
-
-      # Compile in graph plotting code (Not fully implemented)
-      USE_PLOT = true ;		# [true]
-
-      JPEGLIB = ;
-      JPEGINC = ;
-      HAVE_JPEG = true ;
-
-      TIFFLIB = ;
-      TIFFINC = ;
-      HAVE_TIFF = true ;
-
-      PNGLIB = ;
-      PNGINC = ;
-      HAVE_PNG = true ;
-
-      ZLIB = ;
-      ZINC = ;
-      HAVE_Z = true ;
-
-      SSLLIB = ;
-      SSLINC = ;
-      HAVE_SSL = true ;
-
-      LINKFLAGS +=
-        ${lib.concatStringsSep " " (map (x: "-L${x}/lib") buildInputs)}
-        -lrt -lX11 -lXext -lXxf86vm -lXinerama -lXrandr -lXau -lXdmcp -lXss
-        -ljpeg -ltiff -lpng -lssl ;
-    '';
-  in ''
-    cp ${jamTop} Jamtop
-    substituteInPlace Makefile --replace "-j 3" "-j $NIX_BUILD_CORES"
-    # Remove tiff, jpg and png to be sure the nixpkgs-provided ones are used
-    rm -rf tiff jpg png
-
-    export AR="$AR rusc"
+      cat "$extraJamtopPath" >> Jamtop
```

Comment:
❓ I might be missing something, where is `extraJamtopPath` defined?

Reply:
It's the combination of the `extraJamtop` derivation attribute and `passAsFile = ["extraJamtop"]`.